app/queues/notifications/certificate-expiration-monitor.server.ts

```diff
@@ -0,0 +1,59 @@
+import { Queue } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import type { DomainStatusData } from './record-cert-expiration-data.interface';
+import secrets from '~/lib/secrets.server';
+import { prisma } from '~/db.server';
+import {
+  startCertificateExpirationMonitorWorker,
+  certExpirationQueueName,
+} from './certificate-expiration-monitor-worker.server';
+
+// fetch records with by status
+const getCertificatesByExpiration = async () => {
+  let monthFromToday = new Date();
+  monthFromToday.setMonth(monthFromToday.getMonth() + 1);
+  return await prisma.certificate.findMany({
+    where: {
+      AND: [
+        {
+          validTo: {
+            lte: monthFromToday,
+          },
+          isNotified: false,
```

Comment:
I think we should be able to send multiple notifications before expiration. Sorry if I got it wrong, `isNotified` prop sounds we will send only one notification. I'm not sure what is the right interval to send notifications. I think we can consider `lastNotificationSent` prop to compare if the notification is sent today, it will not send again.

Reply:
That was [my initial thought](https://github.com/DevelopingSpace/starchart/pull/268#issuecomment-1445208861). For the scope of this PR, I set it to a boolean, as we are sending notifications 1 month prior to expiration. However, to make the code extendible, I would reflect this change.